### PR TITLE
[ASCII-1823] Remove unnecessary fx Provide blocks to populate API component dependencies

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

Removed unnecessary `fx.Provide` and dependencies from the API component.

We no longer need this dependencies because their endpoint are being register via value groups 🎉

### CODE REVIEW

File: pkg/cli/subcommands/check/command.go [hunk 1]
```diff
@@ -217,18 +207,14 @@ func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
 				// TODO(components): this is a temporary hack as the StartServer() method of the API package was previously called with nil arguments
 				// This highlights the fact that the API Server created by JMX (through ExecJmx... function) should be different from the ones created
 				// in others commands such as run.
-				fx.Provide(func() flare.Component { return nil }),
+				fx.Supply(optional.NewNoneOption[rcservice.Component]()),
+				fx.Supply(optional.NewNoneOption[rcservicemrf.Component]()),
+				fx.Supply(optional.NewNoneOption[logagent.Component]()),
```
Comment (louis-cqrl) [inline pkg/cli/subcommands/check/command.go:212]: small question why we went from this one `fx.Provide(func() optional.Option[logagent.Component] {
					return optional.NewNoneOption[logagent.Component]()

				}),` to this one: `fx.Supply(optional.NewNoneOption[logagent.Component]()),`

**Review by louis-cqrl — COMMENTED** [review]
Seems good to me, very nice to see, however will wait someone with more xp on this to approve

File: pkg/cli/subcommands/check/command.go (hunk 1 above)
Comment (GustavoCaso) [reply pkg/cli/subcommands/check/command.go:212]: It is shorter and it does the same thing 😄

**Review by jeremy-hanna — APPROVED** [review]
👍 LGTM

**Review by jeremy-hanna — APPROVED** [review]

File: cmd/agent/subcommands/jmx/command.go [hunk 2]
```diff
@@ -147,26 +138,19 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
...
-			fx.Provide(func() dogstatsdServer.Component { return nil }),
-			fx.Provide(func() replay.Component { return nil }),
-			fx.Provide(func() pidmap.Component { return nil }),
-			fx.Provide(func() serverdebug.Component { return nil }),
-			fx.Provide(func() host.Component { return nil }),
-			fx.Provide(func() inventoryagent.Component { return nil }),
-			fx.Provide(func() inventoryhost.Component { return nil }),
-			fx.Provide(func() demultiplexer.Component { return nil }),
-			fx.Provide(func() inventorychecks.Component { return nil }),
-			fx.Provide(func() packagesigning.Component { return nil }),
```
Comment (ogaca-dd) [inline cmd/agent/subcommands/jmx/command.go:163]: 👍

**Pull request merged**